capstone/src/main/java/com/google/sps/servlets/YoutubeServlet.java

```diff
@@ -43,8 +43,14 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String ChannelId = request.getParameter("channelId");
-    String url = URL + ChannelId + "&maxResults=" + NUM_RESULTS + "&key=" + KEY;
-    String output = get(url);
+    String videoId = request.getParameter("videoId");
+    String completeUrl;
+    if (ChannelId != null) {
+      completeUrl = BASE_URL + "&allThreadsRelatedToChannelId=" + ChannelId + "&maxResults=" + NUM_RESULTS + "&key=" + KEY;
+    } else {
+      completeUrl = BASE_URL + "&videoId=" + videoId + "&maxResults=" + NUM_RESULTS + "&key=" + KEY;
+    }
```

Comment:
nit: This could be simplified a bit with the ternary operator

https://www.geeksforgeeks.org/java-ternary-operator-with-examples/

Reply:
addressed in "address chris and daniel nits"